acceptance/tests/tasks/task_purged_from_cache.rb

```diff
@@ -0,0 +1,58 @@
+require 'pxp-agent/task_helper.rb'
+
+test_name 'remove old task from pxp-agent cache' do
+
+  task_name = 'foo'
+  cache_path = '/opt/puppetlabs/pxp-agent/tasks-cache/'
+
+
+  step 'Remove contents of cache' do
+    on(agent, "rm -rf #{cache_path}")
+  end
+
+  step 'Start agent with task-cache-dir-purge-ttl configured' do
+    agents.each do |agent|
+      on(agent, puppet('resource service pxp-agent ensure=stopped'))
+      config_hash = pxp_config_hash_using_puppet_certs(master, agent)
+      config_hash['task-cache-dir-purge-ttl'] = '1d'
+      create_remote_file(agent, pxp_agent_config_file(agent), to_hocon(config_hash))
+      on(agent, puppet('resource service pxp-agent ensure=running'))
+
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  step "Create #{task_name} task on agent hosts" do
+    agents.each do |agent|
+      if agent['platform'] =~ /win/
+        task_body = '@echo %PT_message%'
+      else
+        task_body = "#!/bin/sh\necho $PT_message"
+      end
+
+      @sha256 = create_task_on(agent, task_name, 'init.bat', task_body)
+    end
+  end
+
+  step "Run #{task_name} task on agent hosts to populate cache" do
+    run_successful_task(master, agents, task_name, 'init.bat', @sha256, {:message => 'hello'}) do |stdout|
+      assert_equal('hello', stdout.strip, "Output did not contain 'hello'")
+    end
+  end
+
+  step 'Update access time on file and restart pxp-agent to trigger cache purge' do
+    agents.each do |agent|
+      date = DateTime.now - 4
+      on(agent, "ls -l $(find #{cache_path} -type f )") do |result|
+        @target_file_path = result.stdout.match(/#{Regexp.escape(cache_path)}\S*/)[0]
+      end
+      on(agent, "touch --date #{date} #{@target_file_path}")
```

Comment:
The cleanup check is based on directory mtime, not file. So that's what we should be touching.